src/immaculate/groovy/net/neoforged/gradleutils/immaculate/ImmaculateConfiguration.groovy

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.gradleutils.immaculate
+
+import dev.lukebemish.immaculate.CheckTask
+import dev.lukebemish.immaculate.FormattingWorkflow
+import dev.lukebemish.immaculate.ImmaculateExtension
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+
+/**
+ * This is a separate class to avoid class-loading issues when immaculate is *not* applied to the project.
+ */
+final class ImmaculateConfiguration {
+    private ImmaculateConfiguration() {
+    }
+
+    static void apply(Project project) {
+        final configPath = new File(project.rootDir, '.gradle/formatter-config.xml')
+
+        final extract = project.tasks.register('extractGUImmaculateConfiguration', ExtractImmaculateConfiguration) {
+            it.output.set(configPath)
+        }
+
+        project.tasks.withType(CheckTask).configureEach {
+            it.dependsOn(extract)
+        }
+
+        var immaculate = (ImmaculateExtension) project.extensions.getByName("immaculate")
+
+        immaculate.workflows.register("java") {
+            configure(it, configPath)
+        }
+    }
+
+    static void configure(FormattingWorkflow workflow, File configPath) {
+        workflow.java()
+        workflow.trailingNewline()
+        workflow.noTabs()
+        workflow.googleFixImports()
+        workflow.toggleOff.set('spotless:off')
+        workflow.toggleOn.set('spotless:on')
+        workflow.eclipse {
+            it.version '3.37.0'
+            it.config.set(configPath)
+        }
+
+        // courtesy of diffplug/spotless#240
+        // https://github.com/diffplug/spotless/issues/240#issuecomment-385206606
+        workflow.custom 'noWildcardImports', { String fileContents ->
+            if (fileContents.contains('*;\n')) {
+                throw new GradleException('No wildcard imports are allowed!')
+            }
+        }
+
+        workflow.custom 'noNotNull', { String fileContents ->
+            if (fileContents.contains('@NotNull') || fileContents.contains('@Nonnull')) {
+                throw new GradleException('@NotNull and @Nonnull are disallowed.')
+            }
+        }
+
+        workflow.custom 'jetbrainsNullable', { String fileContents ->
+            fileContents.replace('javax.annotation.Nullable', 'org.jetbrains.annotations.Nullable')
+        }
+    }
```

Comment:
Do we want to start enforcing this for all our projects which use GradleUtils? I feel like these are quite substantive changes to make without at least discussing why we should start doing this, particularly that these were not present in GradleUtils Spotless and there is no way to opt-out from these.

Reply:
The only repositories currently using GU Spotless are FML and the installer. I am fine with these rules for the installer (I do not believe they will even change anything for that repo). FML already has these checks
https://github.com/neoforged/FancyModLoader/blob/05a7b8219228f247036e47be124d0eec933b43de/loader/build.gradle#L65-L76
bar the last one but it only uses JetBrains nullable (I don't believe it has access to the javax one)

Reply:
We should revisit these past choices in the future, and get the general agreement of the team on which annotations to use and have that written down somewhere, particularly if we wish to apply these rules more uniformly throughout our many projects (and ideally, _before_ we start applying them).

For now, given these are carbon copies of the rules from NeoForge, these should be acceptable.